### ISSUE TITLE
CMI ASYNC Chunk 1.12.2

### ISSUE THREAD

**Issue by Heriptik** [Zrips/CMI#8645]
### Description of Issue

Hi, CMI try to load a async chunk after a /rtp command

### Version Information

```shell
CMI : 9.7.0.6
CMILib : 1.4.6.2
Server : Spigot 1.12.2-R0.1
CMI economy : activate Vault : 1.7.3-b
Modules : 56 enabled 3 disabled : votifier, portals, skin
```

### Errors

```shell
[16:41:46] [ForkJoinPool.commonPool-worker-13/WARN] [CatServer]: A Mod/Plugin try to async load chunk, it will be executed safely on the main server thread until return!
[16:41:46] [ForkJoinPool.commonPool-worker-13/WARN] [CatServer]: Please check the stacktrace in debug.log and report the author.
[16:41:46] [ForkJoinPool.commonPool-worker-13/DEBUG] [CatServer]: Try to async load chunk
java.lang.Throwable: null
	at catserver.server.AsyncCatcher.checkAsync(AsyncCatcher.java:24) ~[AsyncCatcher.class:git-CatServer-1.12.2-8d58cdb8]
	at net.minecraftforge.common.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:62) ~[ChunkIOExecutor.class:git-CatServer-1.12.2-8d58cdb8]
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:126) ~[on.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.func_186028_c(ChunkProviderServer.java:96) ~[on.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:143) ~[on.class:?]
	at org.bukkit.craftbukkit.v1_12_R1.CraftWorld.getChunkAt(CraftWorld.java:268) ~[CraftWorld.class:git-CatServer-1.12.2-8d58cdb8]
	at net.Zrips.CMILib.Version.PaperMethods.AsyncChunksSync.getChunkAtAsync(AsyncChunksSync.java:12) ~[?:?]
	at net.Zrips.CMILib.Version.PaperMethods.Environment.getChunkAtAsync(Environment.java:34) ~[?:?]
	at net.Zrips.CMILib.Version.PaperMethods.PaperLib.getChunkAtAsync(PaperLib.java:62) ~[?:?]
	at com.Zrips.CMI.Modules.Teleportations.TeleportHandler.getSnapshot(TeleportHandler.java:89) ~[?:?]
	at com.Zrips.CMI.Modules.Teleportations.TeleportHandler.getSnapshot(TeleportHandler.java:81) ~[?:?]
	at com.Zrips.CMI.Modules.Teleportations.RandomTeleportationManager.lambda$14(RandomTeleportationManager.java:503) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604) [?:1.8.0_292]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1596) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175) [?:1.8.0_292]
```

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by mrfloris** [comment]
There have been some more tweaks on spigot cmi/cmilib release about teleporting and support for older servers.
Can you try to upgrade cmi and cmilib and check if this still happens? Report back pls :) tnx

**Comment by Heriptik** [comment]
I have update to 9.7.11 and i have already this error in debug.log : 

```shell
[05:36:28] [ForkJoinPool.commonPool-worker-3/WARN] [CatServer]: A Mod/Plugin try to async load chunk, it will be executed safely on the main server thread until return!
[05:36:28] [ForkJoinPool.commonPool-worker-3/WARN] [CatServer]: Please check the stacktrace in debug.log and report the author.
[05:36:28] [ForkJoinPool.commonPool-worker-3/DEBUG] [CatServer]: Try to async load chunk
java.lang.Throwable: null
	at catserver.server.AsyncCatcher.checkAsync(AsyncCatcher.java:24) ~[AsyncCatcher.class:git-CatServer-1.12.2-8d58cdb8]
	at net.minecraftforge.common.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:62) ~[ChunkIOExecutor.class:git-CatServer-1.12.2-8d58cdb8]
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:126) ~[on.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.func_186028_c(ChunkProviderServer.java:96) ~[on.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:143) ~[on.class:?]
	at net.minecraft.world.World.func_72964_e(World.java:443) ~[amu.class:?]
	at net.minecraft.world.World.func_175726_f(World.java:438) ~[amu.class:?]
	at net.minecraft.world.World.func_180495_p(World.java:1095) ~[amu.class:?]
	at net.minecraft.world.World.func_147447_a(World.java:1131) ~[amu.class:?]
	at net.minecraft.world.World.func_72901_a(World.java:1114) ~[amu.class:?]
	at com.Zrips.CMI.NBT.v1_12_R1.getBlockLookingAt(v1_12_R1.java:1261) ~[?:?]
	at com.Zrips.CMI.Containers.CMIHitBox.getCollidingdBlock(CMIHitBox.java:424) ~[?:?]
	at com.Zrips.CMI.Containers.CMIHitBox.isEmptySpace(CMIHitBox.java:411) ~[?:?]
	at com.Zrips.CMI.Containers.CMIHitBox.isEmptySpace(CMIHitBox.java:403) ~[?:?]
	at com.Zrips.CMI.Modules.Teleportations.TeleportHandler.lambda$5(TeleportHandler.java:156) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604) [?:1.8.0_292]
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1596) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_292]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175) [?:1.8.0_292]
```

**Comment by mrfloris** [comment]
Ah, it's not spigot or paper, yeh modded / forks aren't officially supported, so perhaps why. Up to zrips to find time to try and fix decade old 1.8 forked instance stuff. I suspect he might be slightly busy with 1.20.5 right now. But, thanks for the additional information, that really helps!

**Comment by Zrips** [comment]
Should be resolved with next update
